### PR TITLE
CXP-623: Refactor log to use a dedicated log service

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

The goal is to create a dedicated logger service for the Events API with named methods to create each kind of log.

This will remove the need to instantiate the “log object” directly and call the json_encode function everywhere and it will improve the testing of the classes that log.

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventSubscriptionLogInterface.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+use Akeneo\Connectivity\Connection\Application\Webhook\Log\EventSubscriptionSendApiEventRequestLog;
+use Akeneo\Platform\Component\EventQueue\EventInterface;
+
+/**
+ * @author Pierre Jolly <pierre.jolly@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface EventSubscriptionLogInterface
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventSubscriptionLogInterface.php:15]:
```suggestion
interface EventsApiLogger
```
Naming suggestion, let's see what others think

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/services.yml [hunk 2]
```diff
@@ -85,3 +85,8 @@ services:
 
     akeneo_connectivity.connection.webhook.sleep:
         class: Akeneo\Connectivity\Connection\Infrastructure\Webhook\Service\Sleep
+
+    akeneo_connectivity.connection.webhook.event_subscription_log:
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/services.yml:89]:
```suggestion
    akeneo_connectivity.connection.webhook.events_api_logger:
```

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventSubscriptionLogInterface.php (hunk 1 above)
Comment (willy-ahva) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventSubscriptionLogInterface.php:16]: Do you see in this interface that we are breaking the SRP? If you type-hint the service you inject in all the classes with this interface, it means you enforce to inject in a class that only needs to log a reach limit error, another class that MUST be able to log a reach limit error AND to log an event build error AND to log an event data builder error AND to ... etc. 
What I want to explain is: I'm ok to gather all the methods to log in to one class (even if I'm not convinced of the real benefit), but I'm not ok to create only one interface to gather them. You should create as many interfaces as you have business cases in order to respect the SRP.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Service/EventSubscriptionLog.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+use Akeneo\Platform\Component\EventQueue\BulkEventInterface;
+use Akeneo\Platform\Component\EventQueue\EventInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @author Pierre Jolly <pierre.jolly@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class EventSubscriptionLog implements EventSubscriptionLogInterface
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Service/EventSubscriptionLog.php:18]: This class should implement 5 interfaces.

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventSubscriptionLogInterface.php (hunk 1 above)
Comment (pierrejolly) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventSubscriptionLogInterface.php:16]: Yes you're right I'll fix this

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Service/EventSubscriptionLog.php (hunk 3 above)
Comment (pierrejolly) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Service/EventSubscriptionLog.php:18]: Yes you're right I'll fix this

**Pull request closed without merging**